enhancements/management-workload-partitioning-api.md

```diff
@@ -0,0 +1,239 @@
+   enhancement, explain why but do not remove the section. This part
+   of the process is enforced by the linter CI job.
+
+The `Metadata` section above is intended to support the creation of tooling
+around the enhancement process.
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The overall feature summary is present in the [Management workload partitioning](management-workload-partitioning.md) feature. This document focuses on how the feature enablement API and its interactions will look like.
+
+## Motivation
+
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+
+### Goals
+
+This feature is about designing a read-only API that will describe the enabled workload partitions (types, classes, etc.). This information is needed for kubelet to start exposing the right resources as well for the admission webhook to know when the pod manipulation is needed.
+
+It is expected this API will be created at the installation process. Either manually or using the Performance addon operator render mode.
+
+The short term goal is to get enough functionality to create the `management` partition where all OpenShift infrastructure could be placed. This partition is cluster wide and only affects pods with certain pre-agreed-upon labels and placed in specific pre-annotated namespaces.
+
+### Non-Goals
+
+- Long term goal of allowing cluster admins to create their own partitions is not part of this proposal for now
+- Day 2 creation of partitions is not part of the proposal
+- Namespace or MCP specific partitions are not part of the proposal
+
+## Proposal
+
+The proposal is to define a new cluster-wide Custom Resource Definition that would describe the allowed partition names in the status section. That way it hints at being a read only object where no user/admin input or modifications are expected.
+
+```yaml
+apiVersion: workloadpartitioning.openshift.io/v1
+kind: WorkloadPartitions
+metadata:
+  # arbitrary name, all objects of this Kind should be processed and merged
+  name: management-partition
+status:
+  # list of strings, defines partition names to be exposed by kubelet
+  globalPartitionNames:
+    - management
+```
+
+To allow for future extensibility and possible multiple sources of workload partition names (coming from customers, the installer, or other operators, etc.), we propose that there might be multiple `WorkloadPartitions` objects injected into the cluster. The expected behavior is that all components would just merge all the defined names together.
```

Comment:
In other places where we have cluster-scoped configuration resources like this we look for 1 well-defined name, often `cluster`. I'm not sure how we decide between 1 and many, though. Maybe in this case many makes sense if we assume there may be multiple creators but that the CRD only has status fields. Do we really anticipate multiple creators?

Reply:
If we allow custom partitions in the future then I would say yes, there might be multiple owners in such case (customer creates one and PAO renders another one). I just did not want to close the door prematurely.